Measurements/ReferenceImplementation.qs

```diff
@@ -115,18 +117,75 @@ namespace Quantum.Kata.Measurements {
         }
         return -1;
     }
-    
-    
+
+
     operation TwoBitstringsMeasurement_Reference (qs : Qubit[], bits1 : Bool[], bits2 : Bool[]) : Int {
         // find the first index at which the bit strings are different and measure it
         let firstDiff = FindFirstDiff_Reference(bits1, bits2);
         let res = M(qs[firstDiff]) == One;
-        
+
         return res == bits1[firstDiff] ? 0 | 1;
     }
-    
-    
-    // Task 1.8. |0...0⟩ state or W state ?
+
+    // Task 1.8. Distinguish two superposition states given by two arrays of bit strings
+    // Inputs:
+    //      1) N qubits in |0...0⟩ state which are guaranteed to be
+    //         in one of the two superposition states described by the given bit strings.
+    //      2) two arrays of bit strings represented as bit strings containing Bool[]s.
+    // Output: 0 if qubits were in the basis state(s) described by the first superposition bit string,
+    //         1 if they were in the basis state(s) described by the second superposition bit string.
+    // Bit values false and true correspond to |0⟩ and |1⟩ states.
+    // The state of the qubits at the end of the operation does not matter.
+    // You are guaranteed that both the arrays of bit strings will have the same length
+    // of either 1, 2 or 4.
+    // You are also guaranteed that each bitstring within each array has the same length.
+    // All bit strings, between both arrays, will differ in at least one bit.
+    // You can use exactly one measurement.
+    // Example: for bit string arrays [[false, true, false], [false, false, true]] and [[true, true, true], [false, true, true]]
+    //          return 0 corresponds to state |010⟩ or |001⟩, and return 1 corresponds to state |111⟩ or |011⟩.
+
+    function CheckSuperposition_Reference (superpositionArray : Bool[][], res : Bool, i : Int) : Int {
```

Comment:
Could you please clarify what is this function supposed to do? It looks like it's counting the number of bit strings that do *not* have res in their i-th position, but I'm pretty sure this can be done much easier and faster, without a mutable array. It can also be modified to check whether all bit strings have res (or not res) in their i-th position, in which case you'll be able to return early, as soon as you find a mismatch (or a match).